daemon/delete.go

```diff
@@ -128,6 +129,15 @@ func (daemon *Daemon) cleanupContainer(container *container.Container, forceRemo
 		return errors.Wrapf(err, "unable to remove filesystem for %s", container.ID)
 	}
 
+	for _, l := range container.HostConfig.Links {
+		_, alias, err := opts.ParseLink(l)
+		if err != nil {
+			return err
+		}
+		fullName := path.Join(container.Name, alias)
+		daemon.releaseName(fullName)
```

Comment:
@cpuguy83 Thanks for your suggestion. I have updated the codes and added a testcase for this PR. PTAL :stuck_out_tongue_winking_eye: